addons/binding/org.openhab.binding.vera2/src/main/java/org/openhab/binding/vera2/internal/discovery/VeraBridgeDiscoveryService.java

```diff
@@ -0,0 +1,142 @@
+        logger.debug("Initializing VeraBridgeDiscoveryService");
+    }
+
+    private void scan() {
+        logger.debug("Starting scan for Vera controller");
+
+        ValidateIPV4 validator = new ValidateIPV4();
+
+        try {
+            Enumeration<NetworkInterface> enumNetworkInterface = NetworkInterface.getNetworkInterfaces();
+            while (enumNetworkInterface.hasMoreElements()) {
+                NetworkInterface networkInterface = enumNetworkInterface.nextElement();
+                if (networkInterface.isUp() && !networkInterface.isVirtual() && !networkInterface.isLoopback()) {
+                    for (InterfaceAddress address : networkInterface.getInterfaceAddresses()) {
+                        if (validator.isValidIPV4(address.getAddress().getHostAddress())) {
+                            String ipAddress = address.getAddress().getHostAddress();
+                            Short prefix = address.getNetworkPrefixLength();
+
+                            logger.debug("Scan IP address for Vera Controller: {}", ipAddress);
+
+                            String subnet = ipAddress + "/" + prefix;
+                            SubnetUtils utils = new SubnetUtils(subnet);
+                            String[] addresses = utils.getInfo().getAllAddresses();
+
+                            for (String addressInSubnet : addresses) {
+                                scheduler.execute(new VeraControllerScan(addressInSubnet));
+                            }
+                        }
+                    }
+                }
+            }
+        } catch (SocketException e) {
+            logger.warn("Error occurred while searching Vera controller ({})", e.getMessage());
+        }
+    }
+
+    public boolean pingHost(String host, int port, int timeout) {
+        try (Socket socket = new Socket()) {
+            socket.connect(new InetSocketAddress(host, port), timeout);
+            return true;
+        } catch (IOException e) {
+            return false; // Either timeout or unreachable or failed DNS lookup.
+        }
+    }
+
+    public class VeraControllerScan implements Runnable {
+        private String ipAddress;
+
+        public VeraControllerScan(String ipAddress) {
+            this.ipAddress = ipAddress;
+        }
+
+        @Override
+        public void run() {
+            if (!pingHost(ipAddress, 3480, 100)) {
+                return;
+            }
+
+            try {
+                URL url = new URL("http://" + ipAddress + ":3480/data_request?id=sdata&output_format=json");
+                HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+
+                if (connection.getResponseCode() == 200) {
+                    ThingUID thingUID = new ThingUID(VeraBindingConstants.THING_TYPE_BRIDGE,
+                            ipAddress.replaceAll("\\.", "_"));
+
+                    // Attention: if is already present as thing in the ThingRegistry

```

Comment:
You can avoid this by an additional interface `ExtendedDiscoveryService` just search for it.